src/lib/utils.js

```diff
@@ -166,3 +166,13 @@ export function arraysEqual(a, b) {
 
 	return true
 }
+
+export function getBatteryDescription(node) {
+	return node.batteryLevels !== null
```

Comment:
Telling the truth, I'm a little bit confused how js handles this. In my case value for endpoint 0 is never assigned and here goes test:

```
Welcome to Node.js v20.3.0.
Type ".help" for more information.
> obj = []
[]
> obj[1] = 99
99
> obj[2] = 99
99
> obj.map((v) => `${v}%`).join(',')
',99%,99%'
> obj
[ <1 empty item>, 99, 99 ]
> Object.entries(obj)
[ [ '1', 99 ], [ '2', 99 ] ]
> Object.values(obj)
[ 99, 99 ]
> Math.min(...obj)
NaN
> Math.min(...Object.values(obj))
99
> obj[0] = null
null
> obj.map((v) => `${v}%`).join(',')
'null%,99%,99%'
> obj
[ null, 99, 99 ]
> Object.values(obj)
[ null, 99, 99 ]
> Math.min(...obj)
0
> Math.min(...Object.values(obj))
0
```

Without changing batteryLevels to object instead of Array minBatteryLevel is shown as `null` in my case and battery levels are shown similar to: `null%,60%,90%`

I've tried following patch and in this case minBatterLevel is shown OK: `60%`, but battery levels are still shown as `null%,60%,90%`:

```
--- a/lib/ZwaveClient.ts
+++ b/lib/ZwaveClient.ts
@@ -231,7 +231,7 @@ const observedCCProps: {
 
                        levels[value.endpoint] = value.value
                        node.batteryLevels = levels
-                       node.minBatteryLevel = Math.min(...levels)
+                       node.minBatteryLevel = Math.min(...Object.values(levels))
 
                        this.emitNodeUpdate(node, {
                                batteryLevels: levels,
diff --git a/src/components/nodes-table/SmartView.vue b/src/components/nodes-table/SmartView.vue
index 795a13d7..b10866ff 100644
--- a/src/components/nodes-table/SmartView.vue
+++ b/src/components/nodes-table/SmartView.vue
@@ -507,7 +507,7 @@ export default {
                        } else {
                                description = Array.isArray(node.batteryLevels)
                                        ? 'All battery levels: ' +
-                                         node.batteryLevels.map((v) => `${v}%`).join(',')
+                                         Object.values(node.batteryLevels).map((v) => `${v}%`).join(',')
                                        : 'Unknown battery level'
                                if (level <= 10) {
                                        icon = mdiBatteryAlertVariantOutline
diff --git a/src/components/nodes-table/nodes-table.js b/src/components/nodes-table/nodes-table.js
index 8360d8dc..035b339b 100644
--- a/src/components/nodes-table/nodes-table.js
+++ b/src/components/nodes-table/nodes-table.js
@@ -275,7 +275,7 @@ export default {
                                label = `${level}%`
                                description = Array.isArray(node.batteryLevels)
                                        ? 'All battery levels: ' +
-                                         node.batteryLevels.map((v) => `${v}%`).join(',')
+                                         Object.values(node.batteryLevels).map((v) => `${v}%`).join(',')
                                        : 'Unknown battery level'
                                if (level <= 10) {
                                        icon = mdiBatteryAlertVariantOutline
```

Probably something happens during transfering data from server to client ?

Reply:
I've updated PR - removed sorting and useless checks.